actions/shared/verify_installation/action.yml

```diff
@@ -0,0 +1,162 @@
+    required: true
+    default: '10s'
+  check_tests:
+    description: 'tests are installed or not (true/false)'
+    required: true
+    default: 'false'
+
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Create folder for artifacts
+      run: mkdir -p artifacts
+      shell: bash
+
+    - name: Wait pod status is ready
+      run: |
+        #WAIT POD STATUS IS READY
+        kubectl wait pod --all --for=condition=Ready -n=${{inputs.namespace}} --timeout 100s
+      shell: bash
+      continue-on-error: true
+
+    - name: Get pods
+      run: |
+        #GET PODS
+        kubectl get pods -n ${{inputs.namespace}}
+        file_name=artifacts/${{inputs.namespace}}_get_pods.txt
+        kubectl get pods -n ${{inputs.namespace}} > $file_name
+      shell: bash
+
+    - name: Get events from namespace
+      run: |
+        #GET EVENTS FROM NAMESPACE
+        file_name=artifacts/${{inputs.namespace}}_events.txt
+        kubectl events -n ${{inputs.namespace}} > $file_name
+      shell: bash
+
+    - name: Get yaml from pods
+      run: |
+        #GET YAML FROM PODS
+        file_name=artifacts/${{inputs.namespace}}_pod_yamls.txt
+        IFS=' ' read -ra pods <<< $(kubectl get pods --no-headers -o custom-columns=":metadata.name" -n ${{inputs.namespace}} | tr $'\n' ' ')
+        echo "Yaml files from pods: " > $file_name
+        for pod in "${pods[@]}"; do
+            echo $pod >> $file_name
+            kubectl get pod "$pod" -n ${{inputs.namespace}} -oyaml >> $file_name
+        done
+      shell: bash
+
+    - name: Get status provisioner logs
+      run: |
+        #GET STATUS PROVISIONER LOGS
+        IFS=' ' read -ra pods <<< $(kubectl get pods --no-headers -o custom-columns=":metadata.name" -n ${{inputs.namespace}} | tr $'\n' ' ')
+        for pod in "${pods[@]}"; do
+          if [[ $pod == *"provisioner"* ]]; then
+            provisioner_pod=$pod
+          fi
+        done
+        if [[ $provisioner_pod ]]; then
+          echo "provisioner_pod=$provisioner_pod"
+          for i in {1..${{inputs.max_attempts}}}; do
+            status=$(kubectl get pod $provisioner_pod -n ${{inputs.namespace}} -o jsonpath="{.status.phase}")
+            if [[ "$status" == 'Succeeded' ]]; then
+              break
+            fi
+            sleep ${{inputs.timeout}}
+          done
+          if [[ "$status" != 'Succeeded' ]]; then
+            echo $provisioner_pod pod has not Succeeded status
+          fi
+          echo PROVISIONER LOGS:
+          kubectl logs $provisioner_pod -n ${{inputs.namespace}}
+          logs=$(kubectl logs $provisioner_pod -n ${{inputs.namespace}})
+          if [[ $logs != *"Failed components statuses are []"* ]]; then
+            echo Not all components are in succeed status after ${{inputs.max_attempts}} attempts
+            echo PODS:
+            kubectl get pods -n ${{inputs.namespace}}
+            exit 1
+          fi
+        else
+          echo provisioner pod is not found
+        fi
```

Comment:
It would be better to put the **shorter** then-else branch first. Currently, it is hard to relate `else echo provisioner pod is not found` to the corresponding `if`.

```
if ! [[ $provisioner_pod ]]; then
  echo provisioner pod is not found
  exit 0
fi
```

It would reduce the nesting, and it would make the code easier to follow.

Reply:
fixed